lib/controllers/embed.js

```diff
@@ -0,0 +1,76 @@
+'use strict';
+
+var jwt = require('jsonwebtoken');
+
+var winston = require.main.require('winston');
+var nbbAuthController = require.main.require('./src/controllers/authentication');
+var meta = require.main.require('./src/meta');
+var User = require.main.require('./src/user');
+
+
+var embedControllers = module.exports;
+
+
+embedControllers.embedView = function (req, res, next) {
+	meta.settings.get('openedx-discussion', function (err, settings) {
+		if (err) {
+			return next({
+				code: 'error',
+				plugin: 'openedx-discussion',
+				message: 'Settings could not be loaded',
+			});
+		}
+		var message = '';
+		if (!settings.hasOwnProperty('secret') || !settings.secret.length) {
+			message += '[nodebb-plugin-openedx-discussion] "secret" settings is not set.';
+			winston.error(message);
+		}
+		if (!settings.hasOwnProperty('jwtCookieName') || !settings.jwtCookieName.length) {
+			message = '[nodebb-plugin-openedx-discussion] "jwtCookieName" settings is not set.';
+			winston.error(message);
+		}
+		if (message.length) {
+			return next({
+				code: 'error',
+				plugin: 'openedx-discussion',
+				message: 'nodebb-plugin-openedx-discussion is not configured poroperly',
```

Comment:
If you are logging a message above, shouldn't it be reasonable to pass the same message forward?